### PR TITLE
#14164. Track the most recently accessed chat link

### PR DESCRIPTION
**Pull request by jgandres**

### CODE REVIEW

**Comment by jgandres** [comment]
trigger compilation

**Comment by jgandres** [comment]
trigger compilation

File: include/megaapi.h [hunk 1]
```diff
@@ -6845,6 +6845,13 @@ class MegaApi
             BUSINESS_STATUS_GRACE_PERIOD = 2
         };
 
+        enum {
+            AFFILIATE_TYPE_INVALID = 0, // legacy mode
+            AFFILIATE_TYPE_ID = 1,
+            AFFILIATE_TYPE_FILE_FOLDER_LINK = 2,
+            AFFILIATE_TYPE_CHAT_LINK = 3
```
Comment (sergiohs84) [inline include/megaapi.h]: I suggest to refactor to:
```suggestion
            AFFILIATE_TYPE_FILE_FOLDER = 2,
            AFFILIATE_TYPE_CHAT = 3,
```

File: include/megaapi.h [hunk 2]
```diff
@@ -9975,6 +9982,30 @@ class MegaApi
          */
         void getPaymentId(MegaHandle productHandle, MegaHandle lastPublicHandle, MegaRequestListener *listener = NULL);
 
+        /**
+         * @brief Get the payment URL for an upgrade
+         *
+         * The associated request type with this request is MegaRequest::TYPE_GET_PAYMENT_ID
+         * Valid data in the MegaRequest object received on callbacks:
+         * - MegaRequest::getNodeHandle - Returns the handle of the product
```
Comment (sergiohs84) [inline include/megaapi.h]: We should document the rest of the request params available on all callbacks (ph, pht, phts). Same for `submitPurchaseReceipt`

**Pull request merged**